internal/pkg/manifest/validate.go

```diff
@@ -1937,36 +1948,54 @@ func populateNLBPortsAndValidate(containerNameFor map[uint16]string, opts valida
 		return nil
 	}
 	nlb := opts.nlb
-	if nlb.Port == nil {
+	if nlb.MainListener.Port == nil {
 		return nil
 	}
-	nlbPort, _, err := ParsePortMapping(nlb.Port)
+
+	for _, listener := range nlb.NLBListeners() {
+		if err := populateAndValidateNLBPorts(listener, containerNameFor, opts); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func populateAndValidateNLBPorts(listener NetworkLoadBalancerListener, containerNameFor map[uint16]string, opts validateExposedPortsOpts) error {
+	nlbPort, _, err := ParsePortMapping(listener.Port)
 	if err != nil {
 		return err
 	}
 	port, err := strconv.ParseUint(aws.StringValue(nlbPort), 10, 16)
 	if err != nil {
 		return err
 	}
-	containerPort := uint16(port)
-	if nlb.TargetPort != nil {
-		containerPort = uint16(aws.IntValue(nlb.TargetPort))
+	targetPort := uint16(port)
+	if listener.TargetPort != nil {
+		targetPort = uint16(aws.IntValue(listener.TargetPort))
 	}
-	if exposed, ok := containerNameFor[containerPort]; ok {
-		if nlb.TargetContainer != nil && aws.StringValue(nlb.TargetContainer) != exposed {
-			return &errContainersExposingSamePort{
-				firstContainer:  aws.StringValue(nlb.TargetContainer),
-				secondContainer: exposed,
-				port:            containerPort,
-			}
-		}
+	if err = errOnContainersExposingSamePort(containerNameFor, uint16(aws.IntValue(listener.TargetPort)), listener.TargetContainer); err != nil {
+		return err
 	}
-	targetContainerName := opts.mainContainerName
-	if nlb.TargetContainer != nil {
-		targetContainerName = aws.StringValue(nlb.TargetContainer)
+	targetContainer := opts.mainContainerName
+	if listener.TargetContainer != nil {
+		targetContainer = aws.StringValue(listener.TargetContainer)
 	}
-	containerNameFor[containerPort] = targetContainerName
+	containerNameFor[targetPort] = targetContainer
+	return nil
+}
 
+func errOnContainersExposingSamePort(containerNameFor map[uint16]string, targetPort uint16, targetContainer *string) error {
```

Comment:
nit: maybe something along the line
```suggestion
func validatePortExposedOnce(containerNameFor map[uint16]string, targetPort uint16, targetContainer *string) error {
```